hermes-console/static/js/console/constraints/ConstraintsRepository.js

```diff
@@ -0,0 +1,41 @@
+var repository = angular.module('hermes.constraints.repository', []);
+
+repository.factory('ConstraintsRepository', ['DiscoveryService', '$resource',
+    function (discovery, $resource) {
+        var workloadConstraints = $resource(discovery.resolve('/workload-constraints'), {}, { query: { method: 'GET' } });
+
+        return {
+            getWorkloadConstraints: function () {
+                return workloadConstraints.query({}).$promise
+                    .then(function (response) {
+                        var workloadConstraints = {};
+                        workloadConstraints.topicConstraints = [];
+                        for (topicName in response.topicConstraints) {
+                            if (response.topicConstraints.hasOwnProperty(topicName)) {
```

Comment:
How about iterating with `for (topicName in Object.keys(response.topicConstraints)) {...}`

Reply:
`for (topicName in Object.keys(response.topicConstraints)) {...}` iterates over indexes instead of keys. `for ... of Object.keys(...)` works but not in ES5.
It could be done as `Object.keys(...).forEach(function ...`, but I think it's less readable.

Reply:
is the `hasOwnProperty` check still required?

Reply:
I removed it.